Common/cpp/Registries/EventHandlerRegistry.cpp

```diff
@@ -1,27 +1,49 @@
 #include "EventHandlerRegistry.h"
 #include "WorkletEventHandler.h"
 
+#include <utility>
+
 namespace reanimated {
 
 void EventHandlerRegistry::registerEventHandler(
     std::shared_ptr<WorkletEventHandler> eventHandler) {
   const std::lock_guard<std::mutex> lock(instanceMutex);
   const auto &eventName = eventHandler->getEventName();
   auto handlerId = eventHandler->getHandlerId();
-  eventMappings[eventName][handlerId] = eventHandler;
+
+  if (eventHandler->ignoreEmitterReactTag()) {
+    eventMappings[eventName][handlerId] = eventHandler;
+  } else {
+    const auto emitterReactTag = eventHandler->getEmitterReactTag();
+    const auto eventHash = std::make_pair(emitterReactTag, eventName);
+    eventMappingsWithTag[eventHash][handlerId] = eventHandler;
+  }
   eventHandlers[handlerId] = eventHandler;
 }
 
 void EventHandlerRegistry::unregisterEventHandler(uint64_t id) {
   const std::lock_guard<std::mutex> lock(instanceMutex);
   auto handlerIt = eventHandlers.find(id);
   if (handlerIt != eventHandlers.end()) {
-    const auto &eventName = handlerIt->second->getEventName();
-    auto eventMappingIt = eventMappings.find(eventName);
-    auto &handlersMap = eventMappingIt->second;
-    handlersMap.erase(id);
-    if (handlersMap.empty()) {
-      eventMappings.erase(eventMappingIt);
+    const auto eventHandler = handlerIt->second;
+    const auto &eventName = eventHandler->getEventName();
+
+    if (eventHandler->ignoreEmitterReactTag()) {
+      auto eventMappingIt = eventMappings.find(eventName);
+      auto &handlersMap = eventMappingIt->second;
```

Comment:
Can we throw `const`s here as well?
```suggestion
      const auto eventMappingIt = eventMappings.find(eventName);
      const auto &handlersMap = eventMappingIt->second;
```

Reply:
We can only make `eventMappingIt` const since we modify `handlersMap` later.